### ISSUE TITLE
Docs: Improve node_modules collapsing

### ISSUE THREAD

**Issue by connorjs** [sverweij/dependency-cruiser#755]
### Summary

The docs (and `--init` workflow IIRC) use the following pattern for collapsing `node_modules`.

```
// collapse onto folders one step below node_modules:
collapsePattern: "^(node_modules/[^/]+)",
```

During my own exploration* of dependency cruiser, I found what I believe is a “better” regex for node modules that respects scoped packages.

```
"node_modules/(@[^/]+/[^/]+|[^/]+)"
```

_\*I’m new to this, but oh wow I love it! Dense, but powerful. Great work to everyone involved._

### Context

When I was getting started, it bothered me that I had `@foo` in my collapsed reports instead of `@foo/bar` and `@foo/baz`. Simply adding another level would not work well for `qux` (unscoped) dependencies.

After trying some different regular expressions*, I arrived at the above and was very happy with the results. I now saw `@foo/bar`, `@foo/baz`, and `qux` exactly as mentioned in `package.json` dependencies (with the `@foo`s grouped together). I wanted to open this issue to hear thoughts on this, including any performance concerns if we updated the documentation to “recommend” (default) to this. I would be happy to help update locations if we want it.

\*For example, using `node_modules/(@[^/]+)?/[^/]+"` (`@scope` optional) failed as a potentially expensive regex.

### Environment

- Version used: N/A
- Node version: N/A
- Operating System and version: N/A
- Link to your project: N/A

If it helps, I can create a sample repository with examples. An example image follows.

<img width="184" alt="Example screenshot showing the node_modules grouping with the proposed regular expression. It shows both scoped and unscoped packages. The scoped packages are grouped under their scope." src="https://user-images.githubusercontent.com/22208536/224496558-50ea38db-b134-416c-aac1-c275f3322ce1.png">

**Comment by sverweij** [comment]
Hi @connorjs thanks for this suggestion and the effort you put into this question! I've applied the regular expression into the `--init` template and added it to the documentation in some select spots.

Regarding performance concerns: The collapse pattern is typically applied in the reporterOptions section of the configuration, and as such can only have affect on the speed at which said reporter operates (in this case the `dot` one). I've measured against dependency-cruiser's self scan (on a MBP 2016 with a 2,6 GHz Quad-Core Intel Core i7 with the energy saver engaged), and found the following:

- there is a performance impact of running a reporter with or without a collapse pattern (reporting stage takes ~58ms without vs ~75ms with the pattern).
- the difference between using `^node_modules/[^/]+` and `^node_modules/(@[^/]+/[^/]+|[^/]+)` is not significant (with both the reporter stage takes ~75ms).
